src/climax/climate_projection/arch.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+
+import numpy as np
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+# --------------------------------------------------------
+# References:
+# timm: https://github.com/rwightman/pytorch-image-models/tree/master/timm
+# DeiT: https://github.com/facebookresearch/deit
+# --------------------------------------------------------
+import torch
+import torch.nn as nn
+from climax.arch import ClimaX
+from climax.utils.pos_embed import get_1d_sincos_pos_embed_from_grid
+
+
+class ClimaXClimateBench(ClimaX):
+    def __init__(
+        self,
+        default_vars,
+        out_vars,
+        img_size=[32, 64],
+        time_history=1,
+        patch_size=2,
+        embed_dim=1024,
+        depth=8,
+        decoder_depth=2,
+        num_heads=16,
+        mlp_ratio=4.0,
+        drop_path=0.1,
+        drop_rate=0.1,
+        parallel_patch_embed=False,
+        freeze_encoder=False,
+    ):
+        assert out_vars is not None
+
+        super().__init__(
+            default_vars,
+            img_size,
+            patch_size,
+            embed_dim,
+            depth,
+            decoder_depth,
+            num_heads,
+            mlp_ratio,
+            drop_path,
+            drop_rate,
+            parallel_patch_embed
+        )
+
+        self.out_vars = out_vars
+        self.time_history = time_history
+        self.freeze_encoder = freeze_encoder
+
+        # used to aggregate multiple timesteps in the input
+        self.time_pos_embed = nn.Parameter(torch.zeros(1, time_history, embed_dim), requires_grad=True)
+        self.time_agg = nn.MultiheadAttention(embed_dim, num_heads, batch_first=True)
+        self.time_query = nn.Parameter(torch.zeros(1, 1, embed_dim), requires_grad=True)
+
+        # initialize time embedding
+        time_pos_embed = get_1d_sincos_pos_embed_from_grid(self.time_pos_embed.shape[-1], np.arange(self.time_history))
+        self.time_pos_embed.data.copy_(torch.from_numpy(time_pos_embed).float().unsqueeze(0))
+
+        # overwrite ClimaX
+        # use a linear prediction head for this task
+        self.head = nn.Linear(embed_dim, img_size[0]*img_size[1])
+
+        if freeze_encoder:
+            for name, p in self.blocks.named_parameters():
+                name = name.lower()
+                if 'norm' in name:
+                    continue
+                else:
+                    p.requires_grad_(False)
```

Comment:
Useful to add a shot comment/note here, pointing out that we don't freeze the norm layers.